packages/@lwc/shared/src/html-attributes.ts

```diff
@@ -72,50 +73,39 @@ export function isGlobalHtmlAttribute(attrName: string): boolean {
     return GLOBAL_ATTRIBUTE.has(attrName);
 }
 
-const HTML_ATTRIBUTES_TO_PROPERTY: Record<string, string> = {
-    accesskey: 'accessKey',
-    readonly: 'readOnly',
-    tabindex: 'tabIndex',
-    bgcolor: 'bgColor',
-    colspan: 'colSpan',
-    rowspan: 'rowSpan',
-    contenteditable: 'contentEditable',
-    crossorigin: 'crossOrigin',
-    datetime: 'dateTime',
-    formaction: 'formAction',
-    ismap: 'isMap',
-    maxlength: 'maxLength',
-    minlength: 'minLength',
-    novalidate: 'noValidate',
-    usemap: 'useMap',
-    for: 'htmlFor',
+const HTML_PROPERTIES_TO_ATTRIBUTE: Record<string, string> = {
+    accessKey: 'accesskey',
+    readOnly: 'readonly',
+    tabIndex: 'tabindex',
+    bgColor: 'bgcolor',
+    colSpan: 'colspan',
+    rowSpan: 'rowspan',
+    contentEditable: 'contenteditable',
+    crossOrigin: 'crossorigin',
+    dateTime: 'datetime',
+    formAction: 'formaction',
+    isMap: 'ismap',
+    maxLength: 'maxlength',
+    minLength: 'minlength',
+    noValidate: 'novalidate',
+    useMap: 'usemap',
+    htmlFor: 'for',
 };
 
-const HTML_PROPERTIES_TO_ATTRIBUTE: Record<string, string> = {};
-keys(HTML_ATTRIBUTES_TO_PROPERTY).forEach((attrName) => {
-    const propName = HTML_ATTRIBUTES_TO_PROPERTY[attrName];
-    HTML_PROPERTIES_TO_ATTRIBUTE[propName] = attrName;
-});
+const CAPS_REGEX = /[A-Z]/g;
 
-export function htmlAttributeToProperty(attrName: string): string {
-    if (attrName in AriaAttrNameToPropNameMap) {
-        return AriaAttrNameToPropNameMap[attrName];
-    }
-
-    if (hasOwnProperty.call(HTML_ATTRIBUTES_TO_PROPERTY, attrName)) {
-        return HTML_ATTRIBUTES_TO_PROPERTY[attrName]!;
-    }
-
-    return attrName;
-}
 export function htmlPropertyToAttribute(propName: string): string {
     if (propName in AriaPropNameToAttrNameMap) {
         return AriaPropNameToAttrNameMap[propName];
     }
 
-    if (hasOwnProperty.call(HTML_PROPERTIES_TO_ATTRIBUTE, propName)) {
-        return HTML_PROPERTIES_TO_ATTRIBUTE[propName]!;
+    if (!hasOwnProperty.call(HTML_PROPERTIES_TO_ATTRIBUTE, propName)) {
+        HTML_PROPERTIES_TO_ATTRIBUTE[propName] = StringReplace.call(
```

Comment:
from the code, i would say this computes the attribute name based on the property name (except for those defined in `HTML_PROPERTIES_TO_ATTRIBUTE`).

this way seems reasonable to me, do you know if it is stated as a rule somewhere? i know that for aria attributes is not like this.

Reply:
I believe I've seen a web platform best practice documented somewhere, but reality is different :) 

This change preserves the current behavior so LGTM.

Reply:
this change does not preserves the current behavior; unless im missing something, the current behavior is to return the `propName` if is not in the "exception" list.

Reply:
i think i get what you mean preserving the current behavior @ekashida; i believe `htmlPropertyToAttribute` was not used anywhere, and now is the implementation of the removed `getAttrNameFromPropName`.

Reply:
what will be the output of `htmlPropertyToAttribute ('className')` and `htmlPropertyToAttribute('inputMode')`?

Reply:
I tried to preserve the existing behavior with this PR since the main goal here is to remove the API from `@lwc/engine-core`. Invoking  `htmlPropertyToAttribute('className')` returns `class-name', which is incorrect, however it is also the case on the main branch.

This whole logic is broken and it's certainly something we should move to the compiler.